tools/generators/epic-generator/index.ts

```diff
@@ -0,0 +1,186 @@
+import { getProjects, stripIndents, Tree } from '@nrwl/devkit';
+import { access, readFileSync } from 'fs';
+import { execSync } from 'child_process';
+import fetch from 'node-fetch';
+import { EpicGeneratorSchema } from './schema';
+
+function validateSchema(schema: EpicGeneratorSchema) {
+  if (!schema.repository.match(/[A-z-]+\/[A-z-]+/)) {
+    throw new Error('Must provide a valid repository for the issue');
+  }
+
+  if ((schema.title || '').trim() === '') {
+    throw new Error('Must provide a valid title for the issue');
+  }
+
+  return {
+    repository: schema.repository,
+    title: schema.title,
+    message: schema.message || '*Description to be added*',
+  };
+}
+
+const checkAuthentication = async () => {
+  const authStatus = execSync(`gh auth status`).toString();
```

Comment:
this requires the `gh` client to be installed as a part of the user's setup. Should we do a check that the `gh` command exists first ?

Reply:
also seems that this function is not actually async ?

Reply:
`execSync` already throws a self-explanatory error when the executable is missing, so my thought was that catching said error could cloud transparency when other types of errors are thrown on `execSync`.
We can also do some error management either through text or status code, but both aren't guaranteed to be the same for Windows, Linux, Mac, etc.
Another option would be to add a dependency like [command-exists](https://www.npmjs.com/package/command-exists) and check beforehand.
Wdyt?

Reply:
I don't mind `command-exists` with some more human readable explanation. Below are the error messages I get for WSL and windows:

```
Command gh not found
```

```
gh : The term 'gh' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Not everyone can know what `gh` is especially if said person has moved from working mainly on azure devops or other SCMs

Reply:
I've added handling of `execSync` errors to provide a generic message while also sharing the underlying error. This way we can inform the user to double check their GitHub CLI installation while also maintaining transparency of what's happening underneath. Let me know what you think.

![terminal showing "Error calling GitHub CLI (gh). Please make sure it's installed correctly."](https://user-images.githubusercontent.com/39736248/168030847-ce92772e-fc8f-4f2e-9dd1-9a3b13280b3a.png)